draft-ietf-quic-http.md

```diff
@@ -825,6 +821,70 @@ MUST be treated as a connection error of type HTTP_MALFORMED_FRAME.
 A server MUST treat a MAX_PUSH_ID frame payload that does not contain a single
 variable-length integer as a connection error of type HTTP_MALFORMED_FRAME.
 
+### INITIAL_PRIORITY {#frame-initial-priority}
+
+The INITIAL_PRIORITY (type=0x0F) frame specifies the client-advised priority of
+a new stream.  INITIAL_PRIORITY frames MAY be sent as the first frame of a
+request stream, and MUST NOT be sent thereafter or on other streams.  An
+INITIAL_PRIORITY frame sent on a non-request stream MUST be treated as a
+connection error of type HTTP_WRONG_STREAM.  An INITIAL_PRIORITY frame received
+after other frames or received by a client MUST be treated as a stream error of
+type HTTP_UNEXPECTED_FRAME.
+
+~~~~~~~~~~  drawing
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|DT |   Empty   |           Element Dependency ID (i)         ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|   Weight (8)  |
++-+-+-+-+-+-+-+-+
+~~~~~~~~~~
+{: #fig-initial-priority title="INITIAL_PRIORITY frame payload"}
+
+The INITIAL_PRIORITY frame payload has the following fields:
+
+  Dependency Type:
+  : A two-bit field indicating the type of element being depended on.
```

Comment:
The alignment of the dependency type in comparison to PRIORITY is unfortunate.

Reply:
True -- I considered having a separate two-bit empty field, but that also felt unfortunate.  If you think that's the better of two evils, I'm fine.

Reply:
One alternative would to have a static two-bit field that is always set to the value of `11`. There's no extra overhead since there's unused reserved bits.

However, I like @kazuho's proposal and think there is room to extend it further.

The PRIORITY frame definition says that 

> Element Dependency ID: A variable-length integer that identifies the element on which a dependency is being expressed. Depending on the value of Dependency Type, this contains the Stream ID of a request stream, the Push ID of a promised resource, the Placeholder ID of a placeholder, or is ignored.

IIUC the ignored case properly, wouldn't it be more efficient to make the EDID (chuckle, not [this one](https://en.wikipedia.org/wiki/Extended_Display_Identification_Data)) optional too, based on the prioritisation type. I.e. if the dependency is root of the tree, don't send an ID.

Reply:
That actually could combine nicely -- rather than `11` meaning "the root of the tree," it means "the current stream," and we keep the restriction that reprioritizing the root of the tree is illegal.  So `11` means this stream when it's send on a request, but means the root when it's sent on the control stream.

I was trying not to have conditionally-present fields, but if folks are okay with that, I could try that variant.